app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User'
+
+  def update_comments_count
+    post.update(comments_counter: post.comments.size)
+  end
+end
```

Comment:
- [ ] [OPTIONAL] We can make use of some model callback functions such as __after_save__.

```ruby
  after_save :update_comments_counter

  private

  def update_comments_counter
    author.increment!(:comments_counter) # using the increment function will increment the comments_counter without calling on size function.
  end
```